hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/translators/BlockStreamTransactionTranslator.java

```diff
@@ -0,0 +1,290 @@
+        public static BlockTransaction asBlockTransaction(@NonNull List<BlockItem> items) {
+            final var txn = items.get(0);
+            final var result = items.get(1);
+            final var maybeOutput = items.get(2);
+
+            if (items.size() > 3) {
+                return new BlockTransaction(txn, result, maybeOutput, items.get(3));
+            } else {
+                return new BlockTransaction(txn, result, null, maybeOutput);
+            }
+        }
+    }
+
+    /**
+     * Translates a {@link BlockTransaction} into a {@link TransactionRecord}.
+     *
+     * @param transaction A wrapper for block items representing a single txnInput
+     * @return the translated txnInput record
+     */
+    @Override
+    public TransactionRecord translate(@NonNull final BlockTransaction transaction) {
+        validateBlockItems(transaction.asItems());
+
+        final var recordBuilder = TransactionRecord.newBuilder();
+        final var receiptBuilder = TransactionReceipt.newBuilder();
+
+        BlockItem txnBlockItem = transaction.txnInput();
+        if (txnBlockItem.hasTransaction()) {
+            parseTransaction(txnBlockItem, recordBuilder);
+        }
+
+        txnBlockItem = transaction.transactionResult();
+        if (txnBlockItem.hasTransactionResult()) {
+            parseTransactionResult(txnBlockItem, recordBuilder, receiptBuilder);
+        }
+
+        txnBlockItem = transaction.transactionOutput();
+        if (txnBlockItem != null && txnBlockItem.hasTransactionOutput()) {
+            parseTransactionOutput(txnBlockItem, recordBuilder, receiptBuilder);
+        }
+
+        txnBlockItem = transaction.stateChanges();
+        // TODO: how do we generically extract state changes for the txnInput record?
+        if (txnBlockItem.hasStateChanges()) {}
+
+        return recordBuilder.build();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<TransactionRecord> translateAll(final List<BlockTransaction> transactions) {
+        throw new NotImplementedException();
+    }
+
+    private void validateBlockItems(final List<BlockItem> blockItems) {
+        if (blockItems.size() < 2) {
+            throw new IllegalArgumentException("Expected at least two block items");
+        }
+
+        final var txnItem = blockItems.get(0);
+        if (!txnItem.hasTransaction()) {
+            throw new IllegalArgumentException("Expected first block item to be a txnInput");
+        }
+
+        final var txnResultItem = blockItems.get(1);
+        if (!txnResultItem.hasTransactionResult()) {
+            throw new IllegalArgumentException("Expected second block item to be a txnInput result");
+        }
+
+        final var stateChangesItem = blockItems.getLast();
+        if (!stateChangesItem.hasStateChanges()) {
+            throw new IllegalArgumentException("Expected last block item to be state changes");
+        }
+    }
+
+    private TransactionRecord.Builder parseTransaction(
```

Comment:
(From here on down I'm skipping the `parseXXX` routines because I'm not sure what fields are required to be copied over to the record builders.  For example, here in `parseTransaction`, why only the txid and the memo and not any of the other stuff? Dunno, beyond my scope.)

Reply:
There may be more optimal ways to parse these things. I matched whatever I could find, but probably missed stuff